tools/pinning/pyproject.toml

```diff
@@ -35,6 +35,13 @@ acme = {path = "../../acme", extras = ["dev", "docs"]}
 windows-installer = {path = "../../windows-installer"}
 
 # Extra dependencies
+# We install mock in our "external-mock" tox environment to test that we didn't
+# break Certbot's test API which used to always use mock objects from the 3rd
+# party mock library. We list the mock dependency here so that is pinned, but
+# we don't depend on it in Certbot to avoid installing mock when it's not
+# needed. This dependency can be removed here once Certbot's support for the
+# 3rd party mock library has been dropped.
+mock = "*"
```

Comment:
Does this mean that the version will be updated every time `requirements.txt` is generated? Is that what we want?

Reply:
It will be updated every time `requirements.txt` is generated and a new version of `mock` is available.

I think this is what we want. Prefer the latest version unless we have a reason to pin things back. You can see examples of us pinning things back just below this line in the file. Do you have a preference for something else?

Reply:
Cool, thanks for the explanation. Based on the wording of the PR, I assumed the intention was to not have to deal with updates to mock ever, but I am in fact a fan of getting them periodically, just not quite as often as they're released. This pinning system and its syntax are new to me is why I asked.

Reply:
Totally. [I'm planning to write some more documentation](https://github.com/certbot/certbot/issues/8784) which you may be interested in reading when it's done, but I thought about this pinning problem a lot and am more than happy to answer questions or take feedback.

In general, I think the pinning problem we're trying to solve is a mess that other people don't solve because they don't have to pin their Python software across multiple environments like we do, but I'm relatively happy with the solution I came up with.